src/app/gene-view/gene-view.component.ts

```diff
@@ -775,27 +1039,40 @@ export class GeneViewComponent implements OnInit {
       this.zoomHistory.resetToDefaultState(this.calculateDefaultScale());
       this.setDefaultScale();
     } else {
-      let newXmin: number;
-      let newXmax: number;
+      const x1 = extent[0][0];
+      const x2 = extent[1][0];
 
       // set new frequency limits
       const newFreqLimits = [
         this.convertBrushPointToFrequency(extent[0][1]),
         this.convertBrushPointToFrequency(extent[1][1])
       ];
 
-      if (this.x.domain()[1] - this.x.domain()[0] > 12) {
-        newXmin = Math.round(this.x.invert(extent[0][0]));
-        newXmax = Math.round(this.x.invert(extent[1][0]));
-        if (newXmax - newXmin < 12) {
-          newXmax = newXmin + 12;
-        }
-        this.x.domain([newXmin, newXmax]);
-        this.svgElement.select('.brush').call(this.brush.move, null);
+      let domain: number[];
+      let range: number[];
+      const domainMin = Math.round(this.x.invert(Math.min(x1, x2)));
+      let domainMax = Math.round(this.x.invert(Math.max(x1, x2)));
 
-        this.zoomHistory.append(new GeneViewScaleState(newXmin, newXmax, Math.min(...newFreqLimits), Math.max(...newFreqLimits)));
+      if (domainMax - domainMin < 12) {
+        domainMax = domainMin + 12;
+      }
+      if (this.condenseIntrons) {
+        domain = this.geneViewModel.buildCondensedIntronsDomain(
+          domainMin, domainMax);
+        range = this.geneViewModel.buildCondesedIntronsRange(
+          domainMin, domainMax, this.svgWidth);
+      } else {
+        domain = [domainMin, domainMax];
+        range = [0, this.svgWidth];
       }
 
+      this.x = d3.scaleLinear().domain(domain).range(range).clamp(true);
+
+      if (domainMax - domainMin >= 12) {
+        this.zoomHistory.append(new GeneViewScaleState(domain, range, Math.min(...newFreqLimits), Math.max(...newFreqLimits)));
+      }
```

Comment:
why we put this into if statement?? it seems to me that we have ensured that `domainMax - domainMin >= 12`